makelib/common.mk

```diff
@@ -409,7 +409,29 @@ promote:
 # tag a release
 tag: release.tag
 
-.PHONY: publish.init publish.artifacts publish promote.init promote.artifacts promote tag
+# run code generation
+generate.init: ; @:
+generate.run: ; @:
+generate.done: ; @:
+
+generate:
+	@$(MAKE) generate.init
+	@$(MAKE) generate.run
+	@$(MAKE) generate.done
+
+# prepare for code review
+reviewable:
+	@$(MAKE) generate
+	@$(MAKE) lint
+	@$(MAKE) test
+
+# ensure reviewable target doesn't create a diff
+check-diff: reviewable
+	@$(INFO) checking that branch is clean
+	@test -z "$(shell git status --porcelain)"" || $(FAIL)
```

Comment:
```suggestion
	@test -z "$(shell git status --porcelain)" || $(FAIL)
```

Reply:
Fixed! I also refactored `check-diff` a bit so it won't run the linter or tests needlessly.